torch/utils/hub.py

```diff
@@ -0,0 +1,210 @@
+import importlib
+import os
+import shutil
+import site
+import sys
+import zipfile
+
+if sys.version_info[0] == 2:
+    from urllib2 import urlopen  # noqa f811
+else:
+    from urllib.request import urlopen
+
+import torch
+import torch.utils.model_zoo as model_zoo
+
+KEY_ENTRYPOINTS = 'entrypoints'
+KEY_DEPENDENCIES = 'dependencies'
+KEY_HELP = 'help_msg'
+MASTER_BRANCH = 'master'
+ENV_TORCH_HUB_DIR = 'TORCH_HUB_DIR'
+# TODO: Discussion: default hub_dir
+# DEFAULT_TORCH_HUB_DIR = site.getusersitepackages()
+DEFAULT_TORCH_HUB_DIR = '~/.torch_hub'
```

Comment:
The default directory for models in `torchvision` is `~/.torch/models`. Perhaps, this could be `~/.torch/hub`?